### PR TITLE
WET-410: Created a wet-boew section in GCWeb github

### PR DESCRIPTION
**Pull request by ouafaaetta**
wet-boew folder created with two plugins inside as requested (as a start)

unable to run the "grunt" command

### CODE REVIEW

File: wet-boew/bgimg/index.json-ld [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+	"modified": "2023-09-12",
+	"componentName": "background image",
+	"status": "stable",
+	"pages": {
+		"examples": [
+			{
+				"title": "Background image",
+				"language": "en",
+				"type": "general",
+				"path": "bgimg-en.html"
```
Comment (duboisp) [inline wet-boew/bgimg/index.json-ld:26]: Utilise la propriété "url"

Example : https://github.com/wet-boew/GCWeb/blob/master/common/share/index.json-ld

File: wet-boew/bgimg/bgimg-en.html
Comment (duboisp) [inline wet-boew/bgimg/bgimg-en.html:1]: Utilise cet URL à la place: https://wet-boew.github.io/gcweb-compiled-demos/wetboew-demos/bgimg/bgimg-en.html

File: wet-boew/bgimg/bgimg-doc-en.hbs
Comment (duboisp) [inline wet-boew/bgimg/bgimg-doc-en.hbs:1]: Utilise cet URL à la place de ce fichier: https://wet-boew.github.io/wet-boew/docs/ref/bgimg/bgimg-en.html

**Review by duboisp — CHANGES_REQUESTED** [review]

File: index-en.md [hunk 2]
```diff
@@ -58,6 +59,8 @@ css:
 			<a href="#sitesglobal" class="btn btn-link text-white"><span aria-hidden="true" class="fas fa-globe mrgn-rght-sm"></span>Sites and global</a>
 		</li>
 		<li class="mrgn-lft-md">
+			<a href="#Wet-boew" class="btn btn-link text-white"><span aria-hidden="true" class="fas fa-cube mrgn-rght-sm"></span>Wet-boew</a>
```
Comment (duboisp) [inline index-en.md:62]: L'ancre doit correspondre à l'id ajouté sur l'élément `<h2>` à la ligne 481

 ```suggestion
			<a href="#wetboew" class="btn btn-link text-white"><span aria-hidden="true" class="fas fa-cube mrgn-rght-sm"></span>Wet-boew</a>
```

File: index-en.md [hunk 3]
```diff
@@ -33,6 +33,7 @@ css:
 					<li><a href="#templates"><span aria-hidden="true" class="fas fa-table mrgn-rght-md"></span>Templates</a></li>
 					<li><a href="#experiment"><span aria-hidden="true" class="fas fa-puzzle-piece mrgn-rght-md"></span>Méli-mélo and thematics</a></li>
 					<li><a href="#sitesglobal"><span aria-hidden="true" class="fas fa-globe mrgn-rght-md"></span>Sites and global functionality</a></li>
+					<li><a href="#wet-boew"><span aria-hidden="true" class="fas fa-cube mrgn-rght-md"></span>Wet-boew</a></li>
```
Comment (duboisp) [inline index-en.md:36]: L'ancre doit correspondre à l'id ajouté sur l'élément <h2> à la ligne 481

```suggestion
					<li><a href="#wetboew"><span aria-hidden="true" class="fas fa-cube mrgn-rght-md"></span>Wet-boew</a></li>
```

File: index-en.md [hunk 4]
```diff
@@ -475,7 +478,95 @@ css:
 		</li>
 	{% endfor %}
 	</ul>
-
+<h2 id="wetboew" class="mrgn-bttm-lg">Wet-boew</h2>
+	<ul class="row list-unstyled wb-eqht-grd wb-filter mrgn-tp-md pb-4" data-wb-filter='{ "selector": ">li" }'>
+	{% for wetboew in site.data.wetboew %}
```
Comment (duboisp) [inline index-en.md:483]: Le data file généré contient un tiret, donc le nom de la collection contient aussi un tiret

```suggestion
	{% for wetboew in site.data[ "wet-boew" ] %}
```

File: index-en.md [hunk 5]
```diff
@@ -475,7 +478,95 @@ css:
...
+					{% endif %}
+					<!--
+					Documentation
+					- Link to the documentations if any
+					-->
+					{% if list-pages.docs %}
+					<!--<ul class="list-unstyled mrgn-bttm-lg mrgn-lft-md">-->
+					{% assign docs = list-pages.docs | where: "language", page.language %}
+					{% for doc in docs %}
+						<li><a href="wet-boew/{{ wetboew.componentName }}/{{ doc.path }}"><span class="fas fa-info-circle small mrgn-rght-sm" aria-hidden="true"></span>Documentation</a></li>
```
Comment (duboisp) [inline index-en.md:528]: La logique qui affiche le lien vers la documentation va nécessiter une mise à jour afin qu'il supporte une "url" à la place du "path" une documentation. Une logique similaire lorsqu'on determine le main working example.

Quelques chose similaire à:
```
					{% if mainExamples.path %}
					<a href="wet-boew/
								{%- if wetboew.componentName -%}
									{{ wetboew.componentName }}/
								{%- endif -%}
							{{ mainExamples.path }}" lang="{{ mainExamples.language }}" hreflang="{{ mainExamples.language }}"><span class="fas fa-eye small mrgn-rght-sm" aria-hidden="true"></span>Working example</a>
					{% elsif mainExamples.url %}
						<a href="{{ mainExamples.url }}" lang="{{ mainExamples.language }}" hreflang="{{ mainExamples.language }}"><span class="fas fa-eye small mrgn-rght-sm" aria-hidden="true"></span>Working example</a>
					{% else %}
						<span class="fas fa-eye small mrgn-rght-sm" aria-hidden="true"></span>Working example
					{% endif %}
					{% endif %}
```

File: index-en.md [hunk 6]
```diff
@@ -475,7 +478,95 @@ css:
...
+					{% endfor %}
+					{% endif %}
+					<li><a href="https://github.com/wet-boew/GCWeb/tree/master/components/{{ component.componentName }}" hreflang="en"><span class="fas fa-code small mrgn-rght-sm" aria-hidden="true"></span>Source code</a></li>
```
Comment (duboisp) [inline index-en.md:531]: This base path would need to be adjusted to the new folder location. Like: "tree/master/wet-boew/"

**Review by duboisp — CHANGES_REQUESTED** [review]
I tested in a local jekyll instance and I found a few items. I only commented in English but the French equivalent version might require a similar update.

File: wet-boew/bgimg/index.json-ld [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+		"title": {
+		"en": "Background image",
+		"fr": "Image d'arrière plan"
+	},
+	"description": {
+		"en": "Apply a background image from the given URL in parameter or as per the image group (srcset) information.",
+		"fr": "Défini une image d'arrière plan pour l'URL passé en paramètre ou tel que spécifier par l'information du group d'image (srcset)."
+	},
+	"modified": "2023-09-12",
+	"componentName": "background image",
```
[the remark this answers is not in the document]
Comment (ouafaaetta) [reply wet-boew/bgimg/index.json-ld:18]: done

File: wet-boew/bgimg/index.json-ld (hunk 1 above)
Comment (ouafaaetta) [reply wet-boew/bgimg/index.json-ld:26]: done

File: index-en.md (hunk 6 above)
Comment (ouafaaetta) [reply index-en.md:531]: done

File: wet-boew/bgimg/bgimg-en.html
Comment (ouafaaetta) [reply wet-boew/bgimg/bgimg-en.html:1]: done

File: wet-boew/bgimg/bgimg-doc-en.hbs
Comment (ouafaaetta) [reply wet-boew/bgimg/bgimg-doc-en.hbs:1]: done

**Review by duboisp — CHANGES_REQUESTED** [review]

File: index-en.md (hunk 4 above)
Comment (ouafaaetta) [reply index-en.md:483]: done

File: index-en.md (hunk 5 above)
Comment (ouafaaetta) [reply index-en.md:528]: done

**Review by duboisp — CHANGES_REQUESTED** [review]

File: index-en.md [hunk 8]
```diff
@@ -475,7 +478,101 @@ css:
 		</li>
 	{% endfor %}
 	</ul>
-
+<h2 id="wetboew" class="mrgn-bttm-lg">Wet-boew</h2>
+	<ul class="row list-unstyled wb-eqht-grd wb-filter mrgn-tp-md pb-4" data-wb-filter='{ "selector": ">li" }'>
+	{% for wetboew in site.data.[ "wet-boew" ] %}
```
Comment (duboisp) [inline index-en.md:483]: When I build locally, I got this line throwing the following error:
>     Liquid Warning: Liquid syntax error (line 471): Expected id but found open_square in "wetboew in site.data.[ "wet-boew" ]" in index-en.md

And when I looked at the Github page build log, I found it too.
https://github.com/ouafaaetta/GCWeb/actions/runs/6267244900/job/17019886017#step:4:1248
>     Liquid Warning: Liquid syntax error (line 471): Expected id but found open_square in "wetboew in site.data.[ "wet-boew" ]" in index-en.md

**Review by duboisp — CHANGES_REQUESTED** [review]
See the jekyll build warning/error that are preventing the wet-boew card to be displayed.
Test also the anchor you added, there is one that are not working as expected.

**Review by duboisp — APPROVED** [review]
Review and tested.

Everything look good.

Thanks @ouafaaetta

**Pull request merged**